src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1244,28 +1265,41 @@ MasterExtendedOpNode(MultiExtendedOp *originalOpNode)
 	MultiNode *childNode = ChildNode((MultiUnaryNode *) originalOpNode);
 	MasterAggregateWalkerContext *walkerContext = palloc0(
 		sizeof(MasterAggregateWalkerContext));
+	bool hasNonPartitionColumnDistinctAgg = false;
+	bool repartitionSubquery = false;
 
 	walkerContext->columnId = 1;
-	walkerContext->repartitionSubquery = false;
 
 	if (CitusIsA(parentNode, MultiTable) && CitusIsA(childNode, MultiCollect))
 	{
-		walkerContext->repartitionSubquery = true;
+		repartitionSubquery = true;
 	}
 
+	hasNonPartitionColumnDistinctAgg = HasNonPartitionColumnDistinctAgg(targetEntryList,
+																		originalHavingQual,
+																		tableNodeList);
+
 	/* iterate over original target entries */
 	foreach(targetEntryCell, targetEntryList)
 	{
 		TargetEntry *originalTargetEntry = (TargetEntry *) lfirst(targetEntryCell);
 		TargetEntry *newTargetEntry = copyObject(originalTargetEntry);
 		Expr *originalExpression = originalTargetEntry->expr;
 		Expr *newExpression = NULL;
-
 		bool hasAggregates = contain_agg_clause((Node *) originalExpression);
+
+
```

Comment:
Two new lines. I think we can keep `bool hasAggregates` and `if (hasAggregates)` together.

Reply:
ok